profiling/space-time-stack/kp_space_time_stack.hpp

```diff
@@ -0,0 +1,179 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSTOOLS_PROFILING_SPACETIMESTACK_HPP
+#define KOKKOSTOOLS_PROFILING_SPACETIMESTACK_HPP
+
+#include <chrono>
+
+#include "impl/Kokkos_Profiling_Interface.hpp"
+
+namespace KokkosTools::SpaceTimeStack {
+
+enum Space {
+  SPACE_HOST = 0,
+  SPACE_CUDA = 1,
+  SPACE_HIP  = 2,
+  SPACE_SYCL = 3,
+  SPACE_OMPT = 4
+};
+
+Space get_space(Kokkos::Tools::SpaceHandle const& handle);
+const char* get_space_name(const Space space);
+
+enum { NSPACES = 5 };
```

Comment:
How about you declare it as last enumerator in Space?

Reply:
(to be clear I don't think you should change anything in the tool, In think you should consider a different approach for testing)